api/global/internal/meter.go

```diff
@@ -189,31 +196,50 @@ func (m *meter) newInst(name string, mkind metricKind, nkind core.NumberKind, op
 		opts:  opts,
 	}
 	m.instruments = append(m.instruments, inst)
-	return inst
+	return inst, nil
 }
 
-func newInstDelegate(m metric.Meter, name string, mkind metricKind, nkind core.NumberKind, opts interface{}) metric.InstrumentImpl {
+func delegateCheck(has hasImpl, err error) (metric.InstrumentImpl, error) {
+	if has != nil {
+		return has.Impl(), err
+	}
+	if err == nil {
+		err = metric.ErrSDKReturnedNilImpl
+	}
+	return nil, err
+}
+
+func newInstDelegate(m metric.Meter, name string, mkind metricKind, nkind core.NumberKind, opts interface{}) (metric.InstrumentImpl, error) {
 	switch mkind {
 	case counterKind:
 		if nkind == core.Int64NumberKind {
-			return m.NewInt64Counter(name, opts.([]metric.CounterOptionApplier)...).Impl()
+			return delegateCheck(m.NewInt64Counter(name, opts.([]metric.CounterOptionApplier)...))
```

Comment:
The `opts.([]metric.CounterOptionApplier)` _should_ be fine as the caller should pass the appropriate options with the correct metric kind, but there is no guarantee here and has the chance of panicking.

It seems a bit weird to return a error about wrong type in a (mostly) statically typed language, but seems needed here. Unless, you think it is possible to refactor and use the Go type system?

Reply:
You're right. This is a pre-existing detail, and once we remove the metric-kind-dependent options (monotonic, absolute) as we have planned in OTEP 88, this detail can be fixed to avoid an unsafe type assertion, e.g., https://github.com/open-telemetry/opentelemetry-go/pull/529#discussion_r391184992

I filed #540.